runner_scripts/conan_package_finder.py

```diff
@@ -0,0 +1,325 @@
+        sys.exit(1)
+    
+    # Deduplicate packages
+    deduplicated, version_info = deduplicate_packages(discovered_packages)
+    
+    # Categorize packages
+    cura_package, override_packages = categorize_packages(deduplicated)
+    
+    if args.output_format == "json":
+        result = {
+            "discovered_packages": discovered_packages,
+            "deduplicated_packages": deduplicated,
+            "cura_package": cura_package,
+            "override_packages": override_packages,
+            "version_selections": version_info,
+            "original_count": len(discovered_packages),
+            "deduplicated_count": len(deduplicated)
+        }
+        print(json.dumps(result, indent=2))
+        
+    elif args.output_format == "github-actions":
+        # Output in GitHub Actions format for direct consumption
+        print(f"discovered_packages={' '.join(deduplicated)}")
+        print(f"cura_package={cura_package}")
+        print(f"package_overrides={' '.join(override_packages)}")
+        print(f"original_count={len(discovered_packages)}")
+        print(f"deduplicated_count={len(deduplicated)}")
+        
+    elif args.output_format == "github-summary":
+        # Generate complete GitHub summary and write directly to GITHUB_STEP_SUMMARY
+        jira_ticket = args.jira_ticket or "Unknown"
+        pattern = args.search_pattern or "N/A"
+        
+        summary_lines = [
+            f"## Conan Packages Found for Jira Ticket: {jira_ticket}\n",
+            "\n",
+            f"The workflow searched for Conan packages matching the pattern `{pattern}` across all configured remotes.\n",
+            "\n",
+            "## Discovered Packages:\n",
+            "\n"
+        ]
+        
+        if len(deduplicated) == 0:
+            summary_lines.append("*No packages found matching the specified tag.*\n")
+        else:
+            # Add deduplicated packages list
+            for pkg in deduplicated:
+                summary_lines.append(f"- {pkg}\n")
+            
+            # Add version selection info if deduplication occurred
+            if len(discovered_packages) > len(deduplicated):
+                summary_lines.extend([
+                    "\n",
+                    "### Package Version Selection:\n",
+                    "\n",
+                    f"Found {len(discovered_packages)} total packages, selected {len(deduplicated)} packages after choosing the highest semantic version for each repository.\n",
+                    "\n",
+                    "#### All Discovered Packages (before deduplication):\n",
+                    "\n"
+                ])
+                for pkg in discovered_packages:
+                    summary_lines.append(f"- {pkg}\n")
+        
+        summary_lines.extend(["\n", "---\n"])
+        
+        # Write directly to GitHub Step Summary if available
+        if "GITHUB_STEP_SUMMARY" in environ:
```

Comment:
Recently I have been using an output argument that is set by the script instead. This helps having a Python script that is Github-independant and can be developped/tested individually. Example here: https://github.com/Ultimaker/cura-workflows/blob/825f133c0a0edfe4b04f210830c6d72a6b02f157/runner_scripts/get_conan_broadcast_data.py